feat: OPTIC-1749: Only send 5XX API Exceptions to Sentry

mcanu opened the pull request:
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

This pull requests adds `sentry_skip: True` to API errors that aren't 5XX. We are receiving a lot of errors that are from 4XX, which cause noise in Sentry for us.
Also, removing ...SentryIgnored exceptions.


## Copilot Summary:

This pull request includes multiple changes to the `label_studio` project, primarily focusing on replacing custom exceptions with `ValidationError` from `rest_framework` and updating the exception handling logic. The most important changes are summarized below:

### Exception Handling Updates:

* Replaced `LabelStudioValidationErrorSentryIgnored` with `ValidationError` in the `validate_label_config` function in `label_studio/core/label_config.py`. [[1]](diffhunk://#diff-18a1b8d149cd666039b50f1d108eb67394d16cb88164d908b198fabdf4571034L113-R111) [[2]](diffhunk://#diff-18a1b8d149cd666039b50f1d108eb67394d16cb88164d908b198fabdf4571034L122-R133)
* Updated `custom_exception_handler` in `label_studio/core/utils/common.py` to skip Sentry logging for non-500 `APIException` errors and added handling for `LabelStudioXMLSyntaxErrorSentryIgnored`. [[1]](diffhunk://#diff-0cd678e6cc83f541b3bfc4b09fcfad8c76ee49d47af7c7e41127240bee7b1130L92-R102) [[2]](diffhunk://#diff-0cd678e6cc83f541b3bfc4b09fcfad8c76ee49d47af7c7e41127240bee7b1130R147-R149)

### Code Cleanup:

* Removed unused imports and custom exception classes from `label_studio/core/utils/exceptions.py`. [[1]](diffhunk://#diff-bf1fc4b8b301381ea19a63f97e1fe342090904d68c00fd629e13aca4bdc0fc79L4-R4) [[2]](diffhunk://#diff-bf1fc4b8b301381ea19a63f97e1fe342090904d68c00fd629e13aca4bdc0fc79L30-L45)
* Removed `LabelStudioValidationErrorSentryIgnored` from various files and replaced it with `ValidationError` in `label_studio/data_import/api.py` and `label_studio/projects/models.py`. [[1]](diffhunk://#diff-0d189d758de143ca7197ad41a825d2285df6df57e018672bfe1451dc2785dd3bL416-R415) [[2]](diffhunk://#diff-6ff9748c9ae68f8cccd65f4ee20df06ed0a8b9c32b58da39db3001a7178b1dabL549-R549) [[3]](diffhunk://#diff-6ff9748c9ae68f8cccd65f4ee20df06ed0a8b9c32b58da39db3001a7178b1dabL588-R588) [[4]](diffhunk://#diff-6ff9748c9ae68f8cccd65f4ee20df06ed0a8b9c32b58da39db3001a7178b1dabL614-R614) [[5]](diffhunk://#diff-6ff9748c9ae68f8cccd65f4ee20df06ed0a8b9c32b58da39db3001a7178b1dabL654-R654)

### Import Adjustments:

* Added `ValidationError` import from `rest_framework.exceptions` in `label_studio/core/label_config.py` and `label_studio/projects/models.py`. [[1]](diffhunk://#diff-18a1b8d149cd666039b50f1d108eb67394d16cb88164d908b198fabdf4571034R17-L20) [[2]](diffhunk://#diff-6ff9748c9ae68f8cccd65f4ee20df06ed0a8b9c32b58da39db3001a7178b1dabR48)
* Added `APIException` import in `label_studio/core/utils/common.py`.

### Sentry Configuration:

* Updated `SENTRY_IGNORED_EXCEPTIONS` in `label_studio/core/settings/base.py` to remove exceptions that are now handled by `APIException`.

mcanu committed:
Only log API Exceptions that are 5XX. Replace ...SentryIgnored exceptions

bmartel commented on label_studio/core/settings/base.py:
```diff
@@ -388,16 +388,14 @@
+
+# Exceptions that should not be logged to Sentry and aren't children of drf's APIException class
 SENTRY_IGNORED_EXCEPTIONS = [
     'Http404',
```
What about this one, is it not already ignored from being a non 500? Or is this needed for something else?

mcanu replied:
This exception is from Django and not from DRF, it doesn't inherit from APIException. So it doesn't have a status_code attribute which I can check

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13836150159)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13836150159

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13841113433)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1749-2'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13841113433

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13841113433
